### ISSUE TITLE
ssl value flag is not working as intended

### ISSUE THREAD

**Issue by srikanth4075** [awslabs/amazon-redshift-utils#483]
@IanMeyers I started building the analyze vacuum utility recently and facing the connection issues as shown below:

```
 $docker run --net host  -it 
-e DB=dwhprod 
-e DB_USER=admin_automate 
-e DB_PWD=xxxx 
-e DB_HOST=abcxxx.us-east-1.redshift.amazonaws.com 
-e DB_PORT=5439 
-e DEBUG=True 
-e SSL=True 
-e SCHEMA_NAME=public
-e DO_ANALYZE=True 
amazon-redshift-utils analyze-vacuum
Running analyze-vacuum utility
-- 2020-01-09 21:46:54.447819 [7] Connected to CloudWatch in us-east-1
-- 2020-01-09 21:46:54.447975 [7] Using Cluster Name xxxxxxxxxxxx
-- 2020-01-09 21:46:54.448003 [7] Supplied Args:
{'max_unsorted_pct': '50', 
'max_table_size_mb': '(700*1024)', 
'min_interleaved_skew': '1.4', 
'stats_off_pct': '10', 
'db_pwd': 'xxxx ', 
'min_unsorted_pct': '5', 
'db': 'dwhprod', 
'do_vacuum': True, 
'db_port': 5439, 
'ssl': False,    <=== is set to False even when it is passed as True from docker run
'db_user': 'admin_automate', 
'db_host': 'abcxxx.us-east-1.redshift.amazonaws.com', 
'schema_name': 'public', 
'vacuum_parameter': 'FULL', 
'debug': True, 
'suppress_cw': False,
 'query_slot_count': 1, 
'min_interleaved_count': '0', 
'predicate_cols': False, 
'do_analyze': True}

-- 2020-01-09 21:46:54.448091 [7] Connect abcxxx.us-east-1.redshift.amazonaws.com:5439:dwhprod:admin_automate
Exception on Connect to Cluster: (u'FATAL', u'28000', u'no pg_hba.conf entry for host "::ffff:172.20.101.139", user "admin_automate", database "dwhprod", SSL off', u'/home/ec2-user/padb/src/pg/src/backend/libpq/auth.c', u'466', u'ClientAuthentication')
Unable to connect to Cluster Endpoint
Traceback (most recent call last):
  File "AnalyzeVacuumUtility/analyze-vacuum-schema.py", line 243, in <module>
    main(sys.argv)
  File "AnalyzeVacuumUtility/analyze-vacuum-schema.py", line 234, in main
    result = analyze_vacuum.run_analyze_vacuum(**args)
  File "AnalyzeVacuumUtility/lib/analyze_vacuum.py", line 684, in run_analyze_vacuum
    raise Exception("No Connection was established")
Exception: No Connection was established
```

We have the ssl enabled for the cluster connections. Now we are trying to use docker container to run the analyze and vacuum operations and running into connection issues. We are passing the SSL=True using -e in the docker run command but when looked at Supplied Args it shows as the 'ssl': False' which in turn causing the failure. 

I ran the same from python application itself and it works fine when we pass --require-ssl=True as the argument.

``` 
$python analyze-vacuum-schema.py --db=dwhprod --db-user=admin_automate --db-pwd=xxxx --db-host  abcxxx.us-east-1.redshift.amazonaws.com --require-ssl=True --schema-name public --analyze-flag=True 
-- 2020-01-09 14:03:45.850376 [21979] Connected to CloudWatch in us-east-1
-- 2020-01-09 14:03:46.841349 [21979] [21979] Running 1 out of 1 commands: set statement_timeout = '36000000'
-- 2020-01-09 14:03:47.028953 [21979] Success.
-- 2020-01-09 14:03:47.029066 [21979] [21979] Running 1 out of 1 commands: set application_name to 'AnalyzeVacuumUtility-v.9.1.6'
-- 2020-01-09 14:03:47.213636 [21979] Success.
-- 2020-01-09 14:03:47.213746 [21979] Connected to xxxxxxxx:5439:dwhprod as admin_automate
-- 2020-01-09 14:03:47.213786 [21979] Vacuum flag arg is not set. Vacuum not performed.
-- 2020-01-09 14:03:47.213821 [21979] Warning - Analyze without Vacuum may result in sub-optimal performance
-- 2020-01-09 14:03:47.213855 [21979] Extracting Candidate Tables for analyze based on Query Optimizer Alerts...
-- 2020-01-09 14:04:09.422712 [21979] Found 0 Tables requiring Analysis
-- 2020-01-09 14:04:09.422843 [21979] Extracting Candidate Tables for analyze based on stats off from system table info ...
-- 2020-01-09 14:04:16.884605 [21979] Processing Complete
```

**Comment by raittes** [comment]
@srikanth4075 The env var name in [run-analyze-vacuum-utility](https://github.com/awslabs/amazon-redshift-utils/blob/master/src/bin/run-analyze-vacuum-utility.sh#L18) is `REQUIRE_SSL` instead of `SSL`

**Comment by TheSAS** [comment]
@raittes I have found the issue, and it is here:
https://github.com/awslabs/amazon-redshift-utils/blob/master/src/AnalyzeVacuumUtility/analyze-vacuum-schema.py#L100 
And it is compared with:
https://github.com/awslabs/amazon-redshift-utils/blob/master/src/config_constants.py#L34

And yes, 'require_ssl' != 'ssl' on this line:
https://github.com/awslabs/amazon-redshift-utils/blob/master/src/AnalyzeVacuumUtility/lib/analyze_vacuum.py#L691

**Comment by TheSAS** [comment]
The issue was added by this commit - https://github.com/awslabs/amazon-redshift-utils/commit/6ba151052759758b1dc5244b46d3341a1ee26766#diff-8bee48929f6f5d5dc85a6413f28d1e8f

**Comment by IanMeyers** [comment]
Hopefully this issue with variable naming will be resolved in the 1.5.1 automation package and 1.2.1 of the utility.

**Comment by timjell** [comment]
Closing as issue marked as fixed with deployment
